libraries/AP_Avoidance/AP_Avoidance.h

```diff
@@ -106,6 +106,23 @@ class AP_Avoidance {
     // for holding parameters
     static const struct AP_Param::GroupInfo var_info[];
 
+    // bindings for lua
+    uint8_t num_obstacles() const {
+        return _obstacle_count;
+    }
+    Location get_obstacle_loc(uint8_t instance) const {
+        return _obstacles[instance]._location;
+    }    
+    Vector3f get_obstacle_vel(uint8_t instance) const {
+        return _obstacles[instance]._velocity;
+    }    
+    int32_t get_obstacle_id(uint8_t instance) const {
+        return _obstacles[instance].src_id;
+    }    
+    int32_t get_obstacle_time(uint8_t instance) const {
+        return _obstacles[instance].timestamp_ms;
+    }
+
```

Comment:
I'm thinking it might be better to have this as a single method that returns all these values in a single call right @tpwrules ? 
Apart from the simplicity and flash savings isn't there also the risk of a race condition?